src/protocols/bdx/BDXMessageUtils.h

```diff
@@ -0,0 +1,231 @@
+/*
```

Comment:
Could we change the name to something more focused (which may require some splits)?

I find that sometimes 'utils' or 'helpers' are unavoidable however I would use them as a last resort. For example in current chip we have 'CodeUtils.h' and I would find it more readable to say 'ExitMacros.h' (you can guess by the latter what it actually contains ...).

Is this 'DataStructures.h' or something similar?

This would also force the file descripiton to be more focused (right now 'MessageUtils.h defines structures and utility methods')

Also naming wise, bdx/BDX* stutters. Since I always include this as `#include <protocols/bdx/...>` I would drop the BDX prefix.

Reply:
I agree about dropping the "BDX" from the file name since it's already in the directory name.

The naming for this file is tricky though. It only defines a few structs with methods for helping with packing/unpacking messages (not all messages are complex enough to warrant having a struct defined). I would also argue that `DataStructures.h` is even more ambiguous than `MessageUtils.h`. What about any of the following:
- bdx/MessageStructures.h
- bdx/MessagePacking.h
- bdx/MessageParsing.h
- bdx/MessagePacketBuffer.h

I think MessagePacking or MessageParsing is probably the most descriptive. Are any of these better?

Reply:
MessagePacking sounds good. Better than 'Utils.'

Reply:
What about BdxMessages.h/cpp ? You just remove the "needless" Utils word. It's not utilities, it's the actual messages, so the name fits.

Reply:
Are you suggesting `bdx/BdxMessages.h`? That still suffers from the stuttering problem that Andrei brought up. What about `bdx/Messages.h`? Is that confusing if it doesn't define structures for every message? Something like a BlockAck that only has one field doesn't need a struct definition

Reply:
I think `bdx/BdxMessages.h` is the most correct and descriptive. Tennessee has suggested adding structs for all BDX message types, so this file will cover all messages. And I know you want to avoid the "stutter" problem but I think there will likely be many files that could be `Message.h` so it would be good to differentiate. Also there already exists `echo/EchoClient.h`, `system/SystemEvent.h`, etc

Reply:
For the stutter, we may need to do an additional pass later (we have a ton of CHIP*.h/hpp/cpp files).
If we do not use 'Utils' in the name I am already very happy.